lib/ts/recipe/emaildelivery/types.ts

```diff
@@ -0,0 +1,38 @@
+/* Copyright (c) 2021, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+export interface EmailService<T> {
+    sendEmail: (input: T, userConext: any) => Promise<void>;
+}
+
+export type RecipeInterface<T> = {
+    sendEmail: (input: T, userContext: any) => Promise<void>;
+};
+
+/**
+ * input given by the user if emailDelivery config is passed in parent Recipe
+ */
+export interface TypeConfigInput<T> {
+    service: EmailService<T>;
+    override?: (originalImplementation: RecipeInterface<T>) => RecipeInterface<T>;
+}
+
+/**
+ * config class parameter when parent Recipe create a new EmailDeliveryRecipe object via constructor
+ */
+export interface ConfigInput<T> {
+    service: EmailService<T>;
+    recipeImpl: RecipeInterface<T>;
```

Comment:
You should accept:
```
{
    defaultRecipeImpl: RecipeInterface<T>;
    override: (originalImplementation: RecipeInterface<T>) => RecipeInterface<T>
}
```

This way, this module will apply the override on the defaultRecipeImpl, and all other recipes can just pass the user's config as is (they don't have to apply the override themselves)

Reply:
This comment is wrong, and it's fixed in my PR.